Sources/MapboxNavigation/LaneView.swift

```diff
@@ -1,9 +1,145 @@
+            primaryIndication = .right
+        } else if indications.contains(.uTurn) && maneuverDirection ?? .uTurn == .uTurn {
+            primaryIndication = .uTurn
+        } else {
+            // The lane doesn’t match the maneuver direction, so choose the least extreme indication.
+            // Most likely the lane will appear unhighlighted anyways.
+            if indications.contains(.straightAhead) {
+                primaryIndication = .straightAhead
+            } else if indications.contains(.slightLeft) {
+                primaryIndication = .slightLeft
+            } else if indications.contains(.slightRight) {
+                primaryIndication = .slightRight
+            } else if !indications.isDisjoint(with: .lefts) {
+                primaryIndication = .left
+            } else if !indications.isDisjoint(with: .rights) {
+                primaryIndication = .right
+            } else if indications.contains(.uTurn) {
+                primaryIndication = .uTurn
+            } else {
+                // No indications to draw.
+                return Ranking(primary: [], secondary: nil)
+            }
+        }
+        
+        indications.remove(primaryIndication)
+        
+        // Some dual-use configurations are supported.
+        let secondaryIndication: LaneIndication?
+        if !primaryIndication.isSubset(of: [.straightAhead, .uTurn]) && indications.contains(.straightAhead) {
+            secondaryIndication = .straightAhead
+            
+            // No asset for dual-use slight turn, so use normal turn instead.
+            if primaryIndication == .slightLeft {
+                primaryIndication = .left
+            } else if primaryIndication == .slightRight {
+                primaryIndication = .right
+            }
+        } else if primaryIndication == .straightAhead && !indications.isDisjoint(with: .lefts) {
+            secondaryIndication = .left
+        } else if primaryIndication == .straightAhead && !indications.isDisjoint(with: .rights) {
+            secondaryIndication = .right
+        } else {
+            secondaryIndication = nil
+        }
+        
+        return Ranking(primary: primaryIndication, secondary: secondaryIndication)
+    }
+}
+
+/**
+ A generalized representation of the drawing methods available in LanesStyleKit.
+ */
+enum LaneConfiguration: Equatable {
+    case straightOnly
+    case slightTurnOnly(side: DrivingSide)
+    case turnOnly(side: DrivingSide)
+    case uTurnOnly(side: DrivingSide)
+    
+    case straightOrTurn(side: DrivingSide, straight: Bool, turn: Bool)
+    
+    init?(rankedIndications: LaneIndication.Ranking, drivingSide: DrivingSide) {
+        switch (rankedIndications.primary, rankedIndications.secondary) {
+        // Single-use lanes
+        case (.straightAhead, .none):
+            self = .straightOnly
+        case (.slightLeft, .none):
+            self = .slightTurnOnly(side: .left)
```

Comment:
Then, based on the ranked indications, we choose a generalized “lane configuration” that maps more directly to the set of available drawing methods.